src/main/java/com/github/maxopoly/environmentaleffects/datarepresentations/MobConfig.java

```diff
@@ -142,8 +145,10 @@ public MobConfig(String identifier, EntityType type, String name,
 
 	public Location findSpawningLocation(Location loc) {
 		for (int i = 0; i < maximumTries; i++) {
-			int x = loc.getBlockX() + rng.nextInt(range * 2) - range;
-			int z = loc.getBlockZ() + rng.nextInt(range * 2) - range;
+			int raduis = rng.nextInt(range - minimumDistanceFromPlayer) + minimumDistanceFromPlayer;
+			double angle = rng.nextDouble() * 360;
+			int x = (int) (loc.getBlockX() + (raduis * Math.cos(angle)));
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/lang/Math.html#cos(double)

I missed this @BlackXnt but cos and sin use _radians_ not degrees. 

So change `double angle =` to `double angle = rng.nextDouble() * 2 * Math.PI;`.